src/python/pants/backend/awslambda/python/target_types.py

```diff
@@ -4,9 +4,36 @@
 import re
 from typing import Match, Optional, Tuple, cast
 
-from pants.backend.python.target_types import COMMON_PYTHON_FIELDS, PythonSources
+from pants.backend.python.target_types import PythonInterpreterCompatibility, PythonSources
+from pants.core.goals.package import OutputPathField
 from pants.engine.addresses import Address
-from pants.engine.target import Dependencies, InvalidFieldException, StringField, Target
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    Dependencies,
+    InvalidFieldException,
+    StringField,
+    Target,
+)
+
+
+class DeprecatedPythonAwsLambdaSources(PythonSources):
+    expected_num_files = range(0, 1)
```

Comment:
This is technically changing the API, but things were broken before. This is a much better error message than the error from `MergeDigests`.